### PR TITLE
[#68] [Chore] Write test for LoginUseCase, AuthRepository

### LINKED ISSUE #68: [Chore] Write test for `LoginUseCase`, `AuthRepository`
## Why

As we already added `LoginUseCase`, `AuthRepository`, but we haven't write test for them yet

## Who Benefits?

Developer

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/68

## What happened 👀

Write UnitTest for LoginUseCase and AuthRepository

## Insight 📝

n/a

## Proof Of Work 📹

CI successfully

### CODE REVIEW

**Review by suho — APPROVED** [review]

File: shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        authRepository = AuthRepositoryImpl()
+    }
+
+    @AfterTest
+    fun tearDown() {
+        stopKoin()
+    }
+
+    @Test
+    fun `when login is called - it returns the correctly token`() = runTest {
```
Comment (AVI5HEK) [inline shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt]:
```suggestion
    fun `when login is called - it returns the correct token`() = runTest {
```
or
```suggestion
    fun `when login is called - it returns the token correctly`() = runTest {
```

File: shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        authRepository.saveToken(mockToken)
+
+        verify(mockTokenService)
+            .function(mockTokenService::save)
+            .with(any())
+            .wasInvoked(exactly = 1.times)
+    }
+
+    @Test
+    fun `when getToken is called - it returns the correctly token`() = runTest {
```
Comment (AVI5HEK) [inline shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt]: Same here.
```suggestion
    fun `when getToken is called - it returns the correct token`() = runTest {
```
or 
```suggestion
    fun `when getToken is called - it returns the token correctly`() = runTest {
```

File: shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        "id",
+        "type",
+        "accessToken",
+        "tokenType",
+        1,
+        "refreshToken",
+        1
+    )
+    private val mockToken = AuthToken(mockTokenApi)
+    private val mockThrowable = Throwable("mock")
```
Comment (AVI5HEK) [inline shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt]: Optional: some line breaks between groups would be nice.
```suggestion
    
    @Mock
    private val mockAuthService = mock(classOf<AuthService>())
    @Mock
    private val mockTokenService = mock(classOf<TokenService>())
    
    private lateinit var authRepository: AuthRepository
    
    private val mockTokenApi = AuthTokenApiModel(
        "id",
        "type",
        "accessToken",
        "tokenType",
        1,
        "refreshToken",
        1
    )
    private val mockToken = AuthToken(mockTokenApi)
    private val mockThrowable = Throwable("mock")
```

**Review by AVI5HEK — COMMENTED** [review]
The rest LGTM.

File: shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt (hunk 3 above)
Comment (markgravity) [reply shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt]: @AVI5HEK updated at [560719c](https://github.com/markgravity/kmm-ic/pull/110/commits/560719c698c53127c05b016497d6ef08c4d5b098)

File: shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt (hunk 2 above)
Comment (markgravity) [reply shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt]: @AVI5HEK updated at [560719c](https://github.com/markgravity/kmm-ic/pull/110/commits/560719c698c53127c05b016497d6ef08c4d5b098)

File: shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt (hunk 1 above)
Comment (markgravity) [reply shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/AuthRepositoryTest.kt]: @AVI5HEK updated at [560719c](https://github.com/markgravity/kmm-ic/pull/110/commits/560719c698c53127c05b016497d6ef08c4d5b098)

**Review by phongvhd93 — APPROVED** [review]

**Review by AVI5HEK — APPROVED** [review]

**Pull request merged**